### ISSUE TITLE
Support cached ratelimit

### ISSUE THREAD

**Issue by hzxuzhonghu** [envoyproxy/envoy#14299]
*Title*: *One line description*

*Description*:
>It is better to support caching limit in the client-side instead of calling the rls server for each new request.

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by mattklein123** [comment]
Yeah I've thought about this before and I agree this would be nice to have. One interesting thing to think about would be how to combine the local rate limit filter and global rate limit filter depending on the use case.

**Comment by daixiang0** [comment]
@yanavlasov is it fixed by #16942? I do not find related docs about cached ratelimit.

**Comment by unleashed** [comment]
The current design does not communicate back to the rate limiting service how much of the quota has actually been consumed. Providing this information is key for the service to size the (potentially refreshed) quota it can allocate to a proxy instance at a time, as well as for reporting purposes.

**Comment by yanavlasov** [comment]
I have switched my focus on the new protocol https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/rate_limit_quota_filter

This proposal can still be implemented if needed.

**Comment by nirvanagit** [comment]
@yanavlasov -- wanted to check if the RLQS is under going any new development?
